misc/profiles2/trekking.brf

```diff
@@ -52,7 +54,146 @@ assign is_ldcr =
      if ignore_cycleroutes then false
      else any_cycleroute
 
-assign isbike = or bicycle=yes or or bicycle=permissive bicycle=designated lcn=yes
+#
+# handle one-ways. On primary roads, wrong-not oneways should
+# be close to forbidden, while on other ways we just add
+# 4 to the costfactor (making it at least 5 - you are allowed
+# to push your bike)
+#
+assign badoneway =  # True if we are going against traffic in a one-way street
+       if reversedirection=yes then
+         if oneway:bicycle=yes then true
+         else if oneway= then junction=roundabout
+         else oneway=yes|true|1
+       else oneway=-1
+
+assign goodoneway = if reversedirection=yes then oneway=-1  # True if we are going along traffic in a one-way street
+                      else if oneway= then junction=roundabout else oneway=yes|true|1
+
+assign onewaypenalty =  # Penalty associated with going against traffic in a one-way stret
+       # Note: Can be zero if opposite traffic is allowed for bikes
+       if ( badoneway ) then
+       (
+         if ( cycleway=opposite|opposite_lane|opposite_track ) then 0
+         # No penalty if there is any cycleway in opposite direction
+         # Note: We are already contraflow, so no need for more checks.
+         else if ( cycleway:left=opposite|opposite_lane|opposite_track ) then 0
+         else if ( cycleway:right=opposite|opposite_lane|opposite_track ) then 0
+         # No penalty if shared_busway and busway in opposite direction
+         else if ( and cycleway=share_busway ( or busway=opposite ( or busway:left=opposite busway:right=opposite ) ) ) then 0
+         else if ( and cycleway:left=share_busway ( or busway:left=opposite busway=opposite ) ) then 0
+         else if ( and cycleway:right=share_busway ( or busway:right=opposite busway=opposite ) ) then 0
```

Comment:
I refactored the oneway penalty to correctly take into account the possible contraflow cycleways, in particular the shared busways.